### PR TITLE
Use HPE-MPI v2.25 (1.9 release) for deployment

### PR DESCRIPTION
**Pull request by pramodk**
* we have seen performance regression and deadlocks
   at scale with versions <= 2.23
 * ongoing tests with latest release i.e. v2.25 seems
   promising and we haven't observed deadlock so far.
 * hence, we might need to move to v2.25.

Creating this PR so that we can already have software stack
built with latest MPI and we can test it on reservation.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hpe-mpi/package.py [hunk 1]
```diff
@@ -30,7 +30,10 @@ class HpeMpi(Package):
             url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.21.hmpt.tar.xz')
     version('2.22.hmpt',
             sha256='e067e4ba382d306d540e1ad5bcd63035a4aa93e7d7c15d617e82de4160e5ad8a',
-            url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.22.hmpt.tar.xz',
+            url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.22.hmpt.tar.xz')
+    version('2.25.hmpt',
+            sha256='70ceccca4dccd1865e778b01a631c7e85ea0913d7dbea6608d59ad4be3738187',
+            url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.25.hmpt.tar.xz',
             preferred=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/hpe-mpi/package.py]: Newest version is *always* preferred. If we don't have a non-hmpt version, we don't need this.

Comment (pramodk) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py]: Done

**Comment by matz-e** [comment]
@pramodk ready to go here?

**Comment by olupton** [comment]
It would be nice to move forward with this, as the newer version seems to be (more?) compatible with the clang/sanitizer builds.

**Review by matz-e — DISMISSED** [review]

**Comment by pramodk** [comment]
I think this was mentioned on Slack discussion - HPE told us last week that they should have a new release last Friday. I don't see anything released though. Just asked them again now.

If we don't get anything in a day or two, we will go ahead with deploying this version and then do another update with IME fix.

**Comment by pramodk** [comment]
@sergiorg-hpc can confirm if the updated release has IME fix.

File: var/spack/repos/builtin/packages/hpe-mpi/package.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ class HpeMpi(Package):
             sha256='e067e4ba382d306d540e1ad5bcd63035a4aa93e7d7c15d617e82de4160e5ad8a',
             url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.22.hmpt.tar.xz')
     version('2.25.hmpt',
-            sha256='70ceccca4dccd1865e778b01a631c7e85ea0913d7dbea6608d59ad4be3738187',
+            sha256='126a46bb2cbd4b63bd7b3aed74cee5e8d08e166e9748071fd0b308be29335e1a',
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: Sha change but no version change?

**Review by matz-e — DISMISSED** [review]

File: var/spack/repos/builtin/packages/hpe-mpi/package.py (hunk 2 above)
Comment (sergiorg-hpc) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: @matz-e This is a patch to 2.25. It is technically still 2.25 but ideally with IME fixed. I'm going to test now :)

Comment (matz-e) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: IIRC, this is not going to rebuild anything, though. For that, the version has to change.

Comment (sergiorg-hpc) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: Will this happen even if we run the incubator job trick?

Comment (matz-e) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: Sure, you can do that…

Comment (sergiorg-hpc) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: Done, let me ask Jenkins to redeploy ;)

**Comment by sergiorg-hpc** [comment]
retest this please

**Comment by sergiorg-hpc** [comment]
retest this please

**Comment by sergiorg-hpc** [comment]
retest this please

**Comment by sergiorg-hpc** [comment]
retest this please

**Review by sergiorg-hpc — APPROVED** [review]

**Comment by sergiorg-hpc** [comment]
I expect the CI to finish this time. I will auto-merge the PR, ok?

File: var/spack/repos/builtin/packages/hpe-mpi/package.py (hunk 2 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:35]: > IIRC, this is not going to rebuild anything, though. For that, the version has to change.

Note that:
* in actual deployment we are changing from 2.22 to 2.25
* only my last commit where I changed SHA without version but before doing that I did necessary incubator job trick.

**Comment by matz-e** [comment]
>I expect the CI to finish this time.

why?

**Comment by sergiorg-hpc** [comment]
The CI should be fixed by the following PR:
https://github.com/BlueBrain/spack/pull/1356

**Pull request merged**